pkg/metrics/metrics.go

```diff
@@ -1329,24 +1322,19 @@ func NewLegacyMetrics() *LegacyMetrics {
 		}, []string{
 			LabelSourceCluster,
 			LabelSourceNodeName,
-			LabelTargetCluster,
-			LabelTargetNodeName,
-			LabelTargetNodeType,
 			LabelType,
+			LabelConnectivityStatus,
 		}),
 
-		NodeConnectivityLatency: metric.NewGaugeVec(metric.GaugeOpts{
+		NodeConnectivityLatency: metric.NewHistogramVec(metric.HistogramOpts{
 			ConfigName: Namespace + "_node_connectivity_latency_seconds",
 			Namespace:  Namespace,
 			Name:       "node_connectivity_latency_seconds",
 			Help:       "The last observed latency between the current Cilium agent and other Cilium nodes in seconds",
+			Buckets:    []float64{0.001, 0.0025, 0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1.0, 2.0, 4.0, 8.0},
```

Comment:
I was wondering if these bucket values make sense, http ping has timeout of 30s as far as I can tell:
https://github.com/cilium/cilium/blob/314ca7baeff4f568ffc0bad95124e40665b1f88c/pkg/health/probe/probe.go#L17
While ICMP ping seems to have a default of 60s: https://github.com/cilium/cilium/blob/d20f15ecab7c157f6246a07c857662bec491f6ee/pkg/health/server/server.go#L339

Normally, we would expect latency to be anywhere between X ms - 1s probably.

I think that current bucket values are very reasonable and the problem is mostly in too high timeout for HTTP / icmp probes, which could be addressed during later refactoring in https://github.com/cilium/cilium/issues/32820